### PR TITLE
cmd/fsck: add `repair` option to repair broken directories

### PR DESCRIPTION
**Pull request by SandyXSD**

### CODE REVIEW

File: pkg/meta/redis.go [hunk 1]
```diff
@@ -2824,6 +2824,32 @@ func (m *redisMeta) ListSlices(ctx Context, slices map[Ino][]Slice, delete bool,
 	return errno(err)
 }
 
+func (m *redisMeta) RepairInode(ctx Context, parent, inode Ino) syscall.Errno {
```
Comment (davies) [inline pkg/meta/redis.go:2827]:
```
Check(ctx Context, inode Ino, repair bool) syscall.Errno
```

Comment (SandyXSD) [reply pkg/meta/redis.go:2827]: It's kind of difficult to repair an inode directly.
For example, if the inodeKey is lost, we need parent inode & type information to recover it.

File: cmd/fsck.go [hunk 2]
```diff
@@ -43,18 +43,41 @@ It scans all objects in data storage and slices in metadata, comparing them to s
 lost object or broken file.
 
 Examples:
-$ juicefs fsck redis://localhost`,
+$ juicefs fsck redis://localhost
+
+# Repair broken directories
```
Comment (davies) [inline cmd/fsck.go:48]: it could be a file

File: pkg/meta/base.go [hunk 3]
```diff
@@ -951,6 +953,52 @@ func (m *baseMeta) GetParents(ctx Context, inode Ino) map[Ino]int {
...
+		}
+		if attr.Full {
+			parent = inode
+			continue
+		}
+
+		// missing attribute
+		p := "/" + path.Join(ps[:i+1]...)
+		if attr.Typ != TypeDirectory { // TODO: determine file size?
+			logger.Errorf("Path %s (inode %d type %d) cannot be auto-repaired, you have to repair it manually or remove it", p, inode, attr.Typ)
```
Comment (davies) [inline pkg/meta/base.go:977]: calculate the size based on chunks

**Review by davies — APPROVED** [review]
LGTM for directories. we can fix files with another PR.

**Comment by SandyXSD** [comment]
> LGTM for directories. we can fix files with another PR.

Sure.

**Pull request merged**